components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/utils/RESTAPIAdminClient.java

```diff
@@ -102,21 +108,112 @@ public void updateApi(APITemplateBuilder builder, String tenantDomain) throws Ax
 	}
 
 	/**
-	 * Delete the WebApp from Gateway
+	 * Deletes versioned web app configuration from the gateway
+	 *
 	 * @param tenantDomain
 	 * @throws AxisFault
 	 */
-	public void deleteApi(String tenantDomain) throws AxisFault {
+	public void deleteVersionedWebApp(String tenantDomain) throws AxisFault {
 		try {
 			if (tenantDomain != null && !("").equals(tenantDomain) &&
-			    !tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+					!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
 				restApiAdminStub.deleteApiForTenant(qualifiedName, tenantDomain);
 			} else {
 				restApiAdminStub.deleteApi(qualifiedName);
 			}
-			
+
 		} catch (Exception e) {
 			throw new AxisFault("Error while deleting WebApp", e);
 		}
 	}
+
+	/**
+	 * Adds non-versioned web app configuration to the gateway
+	 *
+	 * @param builder
+	 * @param tenantDomain
+	 * @throws AxisFault
+	 */
+	public void addNonVersionedWebApp(APITemplateBuilder builder, String tenantDomain)
+			throws AxisFault {
+
+		try {
+			String apiConfig = builder.getConfigStringForNonVersionedWebAppTemplate();
+			if (tenantDomain != null && !("").equals(tenantDomain)
```

Comment:
Here null & empty string check is redundant because, when `tenantDomain` equals to `MultitenantConstants.SUPER_TENANT_DOMAIN_NAME` it cannot be null or empty.
Fixed in commit 50d5da1ffa1fffacf500b8005728d72302726392 of PR #153